packages/kotti-ui/source/kotti-field-inline-edit/utils.ts

```diff
@@ -0,0 +1,84 @@
+import {
+	INPUT_LINE_HEIGHT,
+	INPUT_TOP_AND_BOTTOM_PADDING_IN_EM,
+} from './constants'
+import { _ComponentRef } from './types'
+
+export const isFieldTriggered = (
+	fieldRef: _ComponentRef,
+	eventTarget: EventTarget | null,
+) =>
+	fieldRef?.$el === eventTarget ||
+	(eventTarget instanceof HTMLElement && fieldRef?.$el.contains(eventTarget))
+
+export const isInFocus = (component: HTMLElement | null) =>
+	document.activeElement instanceof HTMLElement &&
+	(document.activeElement === component ||
+		component?.contains(document.activeElement))
+
+export const blurField = (fieldRef: _ComponentRef) => {
+	if (
+		document.activeElement instanceof HTMLElement &&
+		isInFocus(fieldRef?.$el ?? null)
+	)
+		document.activeElement.blur()
+}
+
+const increaseHeight = ({
+	inputRef,
+	scrollHeight,
+}: {
+	inputRef: HTMLTextAreaElement | null
+	scrollHeight: number
+}) => {
+	inputRef?.setAttribute('style', `height: ${scrollHeight.toString()}px`)
+}
+
+const decreaseHeight = ({
+	clientHeight,
+	inputRef,
+	value,
+}: {
+	clientHeight: number
+	inputRef: HTMLTextAreaElement | null
+	value: string | null
+}) => {
+	const linesLength = (value ?? '').split('\n').length
+
+	if (inputRef !== null) {
+		const fontSize = Number.parseInt(window.getComputedStyle(inputRef).fontSize)
+
+		const linesHeight = Math.ceil(INPUT_LINE_HEIGHT * fontSize * linesLength)
+		const linesHeightWithPadding =
+			linesHeight + 2 * INPUT_TOP_AND_BOTTOM_PADDING_IN_EM * fontSize
+
+		if (linesHeight < clientHeight) {
+			inputRef.setAttribute(
+				'style',
+				`height: ${linesHeightWithPadding.toString()}px`,
+			)
+		}
+	}
+}
+
+export const resizeTextArea = ({
```

Comment:
I also had to deal with auto-resizing the textarea input but took a different approach, [here](https://github.com/3YOURMIND/kotti/pull/753/files#diff-103fd7d51e3b29bf37e553aec2692d32de50eebb219119089aadac6b233647bb) and [here](https://github.com/3YOURMIND/kotti/pull/753/files#diff-b86df2c207bd01c54cf3cfdc3411c743d16c05b3a61a0dded2af4026a2eed367)

Pretty sure you can do the same unless I'm missing something.

Reply:
@santiagoballadares initially i had just the scrollHeight handling - it didn't work when i needed to decrease the height

Reply:
@carsoli yes, setting `height = scrollHeight` only makes the input grow. 
But if you set the `height = auto` first, it'll decrease its height before increasing it again.

```
export const resizeTextarea = (textarea: HTMLTextAreaElement | null) => {
	textarea?.setAttribute('style', 'height: auto')
	textarea?.setAttribute('style', `height: ${textarea.scrollHeight}px`)
}
```